### PR TITLE
Fix ValueError crash in S3Boto3Storage

### LINKED ISSUE #382: S3Boto3Storage raises ValueError: I/O operation on closed file.
When running `python manage.py collectstatic` we get the following exception:

```
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "~/venvs/app_root/lib/python3.5/site-packages/django/core/management/__init__.py", line 363, in execute_from_command_line
    utility.execute()
  File "items()venvs/app_root/lib/python3.5/site-packages/django/core/management/__init__.py", line 355, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "~/venvs/app_root/lib/python3.5/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "~/venvs/app_root/lib/python3.5/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "~/venvs/app_root/lib/python3.5/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 199, in handle
    collected = self.collect()
  File "~/venvs/app_root/lib/python3.5/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 139, in collect
    for original_path, processed_path, processed in processor:
  File "~/venvs/app_root/lib/python3.5/site-packages/django/contrib/staticfiles/storage.py", line 246, in post_process
    for name, hashed_name, processed, _ in self._post_process(paths, adjustable_paths, hashed_files):
  File "~/venvs/app_root/lib/python3.5/site-packages/django/contrib/staticfiles/storage.py", line 312, in _post_process
    hashed_name = self.hashed_name(name, content_file)
  File "~/venvs/app_root/lib/python3.5/site-packages/django/contrib/staticfiles/storage.py", line 109, in hashed_name
    file_hash = self.file_hash(clean_name, content)
  File "~/venvs/app_root/lib/python3.5/site-packages/django/contrib/staticfiles/storage.py", line 86, in file_hash
    for chunk in content.chunks():
  File "~/venvs/app_root/lib/python3.5/site-packages/django/core/files/base.py", line 76, in chunks
    self.seek(0)
ValueError: I/O operation on closed file.
```

This only happens when using django-storages 1.6.4 or above. 
Versions 1.6.3 and lower work fine.

We're using Django 1.11.4, python 3.5.2, boto3 1.4.6.

### PR DESCRIPTION
**Pull request by WhyNotHugo**
Using this storage with `ManifestFilesMixin`, and in some other
not-so-uncommon scenarios.

This tweak essentially retains the changes from c73680e, but fixes #382.

### CODE REVIEW

**Comment by WhyNotHugo** [comment]
Looks like some tests require updates; looking into it.

**Comment by WhyNotHugo** [comment]
Tests are failing for python 3.3, see #438 for a fix.

File: storages/backends/s3boto3.py [hunk 1]
```diff
@@ -446,8 +446,7 @@ def _save(self, name, content):
         # even if wrapped file-like object exists. To avoid Django-specific
         # logic, pass internal file-like object if `content` is `File`
         # class instance.
-        if isinstance(content, File):
-            content = content.file
+        content.name = name
```
Comment (xrmx) [inline storages/backends/s3boto3.py]: comment does not match code anymore

**Comment by jschneier** [comment]
Merged #754 to fix this instead, thanks for your contribution.

**Pull request closed without merging**